src/app/common/forms/validate.js

```diff
@@ -0,0 +1,43 @@
+import isEmpty from 'lodash/isEmpty'
+export const errors = {
+  required: 'Required',
+  email: 'Email is wrong',
+}
+
+
+export function mainValidation(fields, errorMessage, validate) {
+  return function(values) {
+    if(!Array.isArray(fields)) {
+      fields = [fields]
+    }
+    if(isEmpty(fields)) {
+      throw new Error('fields should be defined')
+    }
+    return fields.reduce(function(res, key) {
+      if(!validate(values[key])) {
+        return {
+          ...res,
+          [key]: errorMessage,
+        }
+      }
+      return res
+    }, {})
+  }
+}
+
+export function compose(...validations) {
+  return function(values) {
+    return validations.reduce(function(errors, validator) {
+      return { ...errors, ...validator(values) }
+    }, {})
+  }
+}
+
+export function required(fields) {
+  return mainValidation(fields, errors.required, (value) => !!value)
+}
+
+export function email(fields) {
+  const emailRe = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-z\u0080-\u00ff\-0-9]+\.)+[a-zA-Z]{2,}))$/
+  return mainValidation(fields, errors.email, (value) => value && emailRe.test(value))
+}
```

Comment:
1. please move each validation function to separate file
2. think about the structure:
```
form/validation/utils/mainValidation.js
form/validation/email.js
form/validation/required.js
form/validation/constants.js
form/validation/index.js 
```